File: typescript-client-sdk/src/sdk/models/operations/findpet.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import * as utils from "../utils";
import { Type } from "class-transformer";


export class FindpetPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}

export class FindpetRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: FindpetPathParams;

  @SpeakeasyMetadata()
  retries?: utils.RetryConfig;
}

export class FindpetResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  pet?: shared.Pet;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/findpet.go
package operations

import (
	"github.com/speakeasy-sdks/nolan-test/pkg/models/shared"
	"github.com/speakeasy-sdks/nolan-test/pkg/utils"
)

type FindpetPathParams struct {
	ID int64 `pathParam:"style=simple,explode=false,name=id"`
}

type FindpetRequest struct {
	PathParams FindpetPathParams
	Retries    *utils.RetryConfig
}

type FindpetResponse struct {
	ContentType string
	Error       *shared.Error
	Pet         *shared.Pet
	StatusCode  int
}
